tests/SelfTest/IntrospectiveTests/CmdLine.tests.cpp

```diff
@@ -569,6 +569,27 @@ TEST_CASE( "Process can be configured on command line", "[config][command-line]"
             REQUIRE(config.benchmarkWarmupTime == 10);
         }
     }
+
+    SECTION("Sharding options") {
+        SECTION("shard-count") {
+            CHECK(cli.parse({ "test", "--shard-count=8"}));
+
+            REQUIRE(config.shardCount == 8);
+        }
+
+        SECTION("shard-index") {
+            CHECK(cli.parse({ "test", "--shard-index=2"}));
+
+            REQUIRE(config.shardIndex == 2);
+        }
+
+        SECTION("Zero shard-count") {
+            auto result = cli.parse({ "test", "--shard-count=0"});
```

Comment:
Without digging too deep, I suspect that providing a negative number would pass, but result in a nonsensically large shard count.

Whether that is or isn't true, there should be a test for it.